src/mon/OSDMonitor.cc

```diff
@@ -1663,6 +1665,15 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
     for (map<int,failure_reporter_t>::iterator p = fi.reporters.begin();
 	 p != fi.reporters.end();
 	 ++p) {
+      //get subtree reporter
+      map<string, string> reporter_loc = osdmap.crush->get_full_location(p->first);
+      map<string, string>::iterator iter = reporter_loc.find(reporter_subtree_level); 
```

Comment:
Please verify that this also includes osd = osd.123 .. I think it doesn't, and we want to make mon_osd_reporter_subtree_level=osd work too.